### PR TITLE
fix: update currency value decimals toFixed(2)

### LINKED ISSUE #2935: Number of decimals for converted non-blockchain currencies
## Steps to reproduce

1. Ensure you have funds in a colony that has a value.
2. Ensure you have a currency selected that is not blockchain based.
3. Note the number formatting.

## Expected behaviour

* All non-blockchain based converted currencies should have 2 decimal places.
* For all non-blockchain based converted currencies we should not abbreviate until over 9 digits.

image

## Actual behaviour

* All non-blockchain based converted currencies, have 5 decimal places. Having more then 2 is not typical number formatting for non-blockchain currencies that people typically use everyday. This makes it harder to read.

image

### PR DESCRIPTION
**Pull request by Nortsova**
- [x] All non-blockchain based converted currencies should have 2 decimal places.
- [x] For all non-blockchain based converted currencies we should not abbreviate until over 9 digits.

image - image - image - image 

## Testing
* Step 1. Open dashboard and see that all more than 10 million is abbreviated

image

* Step 2. Transfer 1,000,000 to Andromeda
* Step 3. Navigate to Andromeda tab and ensure that 1,000,000.00 doesn't abbreviated
image

* Step 4. Pay 1 token from Serenity to any member
* Step 5. Navigate to the dashboard and select the Serenity team
* Step 6. Ensure that the total amount has no more than 2 decimals
image

* Step 7. Transfer 100,000 to Rocinante
* Step 8. Navigate to the dashboard and select the Rocinante team
* Step 9. Ensure that the total amount shown correctly 
image

## New staff:
Moved Numeral UI to NumeralComponent.tsx
Created NumeralCurrency component 

Resolves  #2935

### CODE REVIEW

File: src/components/shared/Numeral/NumeralComponent.tsx
Comment (mmioana) [inline src/components/shared/Numeral/NumeralComponent.tsx:1]: I would rename this into `NumeralBase` to be consistent with the other components. What do you think? 🌟

Comment (Nortsova) [reply src/components/shared/Numeral/NumeralComponent.tsx:1]: thank you! I updated that

File: src/components/v5/frame/ColonyHome/partials/FundsCards/FundsCards.tsx [hunk 1]
```diff
@@ -48,7 +48,7 @@ export const FundsCards = () => {
                   currency={currency}
                   value={
                     <Numeral
-                      value="15,779.00"
+                      value={15779.443}
```
Comment (rdig) [inline src/components/v5/frame/ColonyHome/partials/FundsCards/FundsCards.tsx:51]: I assume this is just temporary

**Review by rdig — APPROVED** [review]
Yep, everything works as expected here, very nice.

I've ran all the testing steps as you described and everything shows up correctly.

Screenshot from 2024-09-25 12-38-38
Screenshot from 2024-09-25 12-39-12
Screenshot from 2024-09-25 12-39-25
Screenshot from 2024-09-25 12-40-21
Screenshot from 2024-09-25 12-40-34
Screenshot from 2024-09-25 12-41-08
Screenshot from 2024-09-25 12-41-21

**Review by mmioana — APPROVED** [review]
Awesome fix and refactoring @Nortsova ✨ 🫶 Thank you for the effort!

Tested and seems the formatting is properly working for these cards
Screenshot 2024-09-25 at 13 41 11
Screenshot 2024-09-25 at 13 41 15
Screenshot 2024-09-25 at 13 41 18
Screenshot 2024-09-25 at 13 41 22
Screenshot 2024-09-25 at 13 41 26
Screenshot 2024-09-25 at 13 42 31
Screenshot 2024-09-25 at 13 42 38

**Review by iamsamgibbs — COMMENTED** [review]
I'm not sure if there are issues with my local environment or something, but I haven't been able to test this properly.

Amounts over 1 million appear properly:

Screenshot 2024-09-26 at 09 26 01

Transferring to Andromeda works as expected:

Screenshot 2024-09-26 at 09 26 44

There is something funky going on with payments though, the amounts aren't adding up properly, here the amount should drop from 65 to ~60 but it instead drops to 57:

https://github.com/user-attachments/assets/5884abf7-a44b-44b8-96c2-07e87630948e

I couldn't get it to display any prefix other than `.00` even when trying to make a payment of `0.42` tokens.

The amount of native CREDs doesn't seem to line up with the currency amount when you have a team selected either (I think in local development 1CRED always equals 1GBP / USD / EUR but I could be wrong)

Screenshot 2024-09-26 at 09 39 10

Looking at the code I don't think any of these issues are necessarily introduced here, but it makes it very hard to test if this is working as expected.

**Comment by mmioana** [comment]
> I'm not sure if there are issues with my local environment or something, but I haven't been able to test this properly.
> 
> Amounts over 1 million appear properly:
> 
> Screenshot 2024-09-26 at 09 26 01
> Transferring to Andromeda works as expected:
> 
> Screenshot 2024-09-26 at 09 26 44
> There is something funky going on with payments though, the amounts aren't adding up properly, here the amount should drop from 65 to ~60 but it instead drops to 57:
> 
>  Screen.Recording.2024-09-26.at.09.34.46.mov 
> I couldn't get it to display any prefix other than `.00` even when trying to make a payment of `0.42` tokens.
> 
> The amount of native CREDs doesn't seem to line up with the currency amount when you have a team selected either (I think in local development 1CRED always equals 1GBP / USD / EUR but I could be wrong)
> 
> Screenshot 2024-09-26 at 09 39 10
> Looking at the code I don't think any of these issues are necessarily introduced here, but it makes it very hard to test if this is working as expected.

@iamsamgibbs I think our PR should fix the issue with `There is something funky going on with payments though, the amounts aren't adding up properly, here the amount should drop from 65 to ~60 but it instead drops to 57`

**Review by iamsamgibbs — APPROVED** [review]
I still can't get any value other than `.00` to show at the end of a payment, but as mentioned before I don't think that is related to any changes here and will hopefully be resolved by other branches getting merged in, so approving this one!

Screenshot 2024-09-26 at 20 15 25
Screenshot 2024-09-26 at 20 16 17
Screenshot 2024-09-26 at 20 17 01
Screenshot 2024-09-26 at 20 17 51

**Comment by mmioana** [comment]
@iamsamgibbs I will open another issue for this, as the lambda might truncate the values 🤔

**Comment by Nortsova** [comment]
Thank you, @iamsamgibbs! I will definitely keep an eye on it after merging with @mmioana  fix, and if it’s still broken, I have an idea of what it could be. 🙌

**Pull request merged**